### ISSUE TITLE
kalign fail when sequences in input file are almost the same.

### ISSUE THREAD

**Issue by xiekunwhy** [TimoLassmann/kalign#43]
Hi,

Kalign fail when sequences in input file are almost the same. Would you please fix this bug.
[family.zip](https://github.com/TimoLassmann/kalign/files/14970745/family.zip)

Best,
Kun

```
/Bio/User/kxie/software/mambaforge/envs/kalign3/bin/kalign --nthreads 4 --format fasta --type protein -i /state/partition1/WORK/Bio/Project/xiek/pipe/compareGenomic/BeiChaiHu/pipe3/02.ortholog/04.msa/pep/OG0000001.pep.fa -o /state/partition1/WORK/Bio/Project/xiek/pipe/compareGenomic/BeiChaiHu/pipe3/02.ortholog/04.msa/alnpep/OG0000001.pep.aln

Kalign (3.4.0)

Copyright (C) 2006,2019,2020,2021,2023 Timo Lassmann

This program comes with ABSOLUTELY NO WARRANTY; for details type:
`kalign -showw'.
This is free software, and you are welcome to redistribute it
under certain conditions; consult the COPYING file for details.

Please cite:
  Lassmann, Timo.
  "Kalign 3: multiple sequence alignment of large data sets."
  Bioinformatics (2019) 
  https://doi.org/10.1093/bioinformatics/btz795

[2024-04-14 20:01:58] :     LOG : Detected protein sequences.
[2024-04-14 20:01:58] : WARNING : -------------------------------------------- (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/msa_op.c line 522)
[2024-04-14 20:01:58] : WARNING : All input sequences have the same length.    (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/msa_op.c line 523)
[2024-04-14 20:01:58] : WARNING : BUT there are no gap characters.             (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/msa_op.c line 524)
[2024-04-14 20:01:58] : WARNING :                                              (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/msa_op.c line 525)
[2024-04-14 20:01:58] : WARNING : Unable to determine whether the sequences    (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/msa_op.c line 526)
[2024-04-14 20:01:58] : WARNING : are already aligned.                         (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/msa_op.c line 527)
[2024-04-14 20:01:58] : WARNING : Kalign will align the sequences.             (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/msa_op.c line 528)
[2024-04-14 20:01:58] : WARNING : -------------------------------------------- (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/msa_op.c line 529)
[2024-04-14 20:01:58] :     LOG : Read 253 sequences from /state/partition1/WORK/Bio/Project/xiek/pipe/compareGenomic/BeiChaiHu/pipe3/02.ortholog/04.msa/pep/OG0000001.pep.fa.
[2024-04-14 20:01:58] :     LOG : CPU Time: 0.00u 00:00:00.00 Elapsed: 00:00:00.00
[2024-04-14 20:01:58] :     LOG : Calculating pairwise distances
[2024-04-14 20:01:59] :     LOG : CPU Time: 0.24u 00:00:00.24 Elapsed: 00:00:01.00
[2024-04-14 20:01:59] :     LOG : Building guide tree.
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:01:59] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
[2024-04-14 20:02:00] :   ERROR : Failed. (/opt/conda/conda-bld/kalign3_1702189157882/work/lib/src/bisectingKmeans.c line 506)
Segmentation fault
```

**Comment by TimoLassmann** [comment]
Hi Kun,

Thanks for reporting! I fixed this issue in the recent commit.

Best, T

**Comment by hangsuUNC** [comment]
Hi Tim,

I got the same error when running kalign. Could you please point me with the released version with the fix? Currently I'm using kalign v3.4.0. 

Thanks!

Hang
